crates/bevy_ecs/src/schedule_v3/executor/mod.rs

```diff
@@ -0,0 +1,95 @@
+mod multi_threaded;
+mod simple;
+mod single_threaded;
+
+pub use self::multi_threaded::MultiThreadedExecutor;
+pub use self::simple::SimpleExecutor;
+pub use self::single_threaded::SingleThreadedExecutor;
+
+use std::cell::RefCell;
+
+use fixedbitset::FixedBitSet;
+
+use crate::{
+    schedule_v3::{BoxedCondition, NodeId},
+    system::BoxedSystem,
+    world::World,
+};
+
+/// Types that can run a [`SystemSchedule`] on a [`World`].
+pub(super) trait SystemExecutor: Send + Sync {
+    fn kind(&self) -> ExecutorKind;
+    fn init(&mut self, schedule: &SystemSchedule);
+    fn run(&mut self, schedule: &mut SystemSchedule, world: &mut World);
+}
+
+/// Specifies how a [`Schedule`](super::Schedule) will be run.
+///
+/// [`MultiThreaded`](ExecutorKind::MultiThreaded) is the default.
+#[derive(PartialEq, Eq, Default)]
+pub enum ExecutorKind {
+    /// Runs the schedule using a single thread.
+    SingleThreaded,
+    /// Like [`SingleThreaded`](ExecutorKind::SingleThreaded) but calls [`apply_buffers`](crate::system::System::apply_buffers)
+    /// immediately after running each system.
+    Simple,
+    #[default]
+    /// Runs the schedule using a thread pool. Non-conflicting systems can run in parallel.
```

Comment:
I think the default should belong below the documentation, shouldn't it? (ok yeah thats a really tiny thing but nvm..)

Reply:
@maniwani this can be marked as resolved now 😄